test/face/scan/test_scan.py

```diff
@@ -0,0 +1,476 @@
+    """
+    def check_label():
+        """
+        Checking the interface elements change asynchronously
+        """
+        assert isinstance(scantest.dialog, QMessageBox)
+        assert scantest.dialog.icon() == QMessageBox.Information
+        assert scantest.dialog.windowTitle() == "Faulty scanning"
+        if isinstance(expt, OSError):
+            assert "Selected executable of Tesseract OCR is unfunctional." in scantest.dialog.text()
+        elif isinstance(expt, TesseractError):
+            assert "Processing failed as either Tesseract OCR executable ceased to function or training data was tampered with." in scantest.dialog.text()
+        assert scantest.dialog.isVisible()
+    qtbot.waitUntil(check_label)
+
+    """
+    Clear the copied file from temp directory
+    """
+    kill_temp_file()
+    if path.exists(savefile):
+        remove(savefile)
+
+
+@pytest.mark.parametrize(
+    "_",
+    [
+        pytest.param(None, id="face.scan.rule: Clearing Snapshot")
+    ]
+)
+def test_scan_snapshot_wipe(scantest, qtbot, _) -> None:
+    """
+    Attempt clearing of the snapshot
+
+    :return:
+    """
+
+    """
+    Perform the action to clear the snapshot
+    """
+    qtbot.mouseClick(scantest.arti_cnvs_wipe, Qt.LeftButton)
+
+    """
+    Confirm if the user interface elements change accordingly
+    """
+    assert scantest.arti_shot.text() == "YOUR ARTIFACT SCREENSHOT WILL SHOW UP HERE"
+
+
+@pytest.mark.parametrize(
+    "_",
+    [
+        pytest.param(None, id="face.scan.rule: Importing artifact information in MainWindow")
+    ]
+)
+def test_scan_import_arti(scantest, qtbot, mocker, _) -> None:
+    """
+    Attempt importing artifact information in MainWindow
+
+    :return:
+    """
+
+    """
+    Create the tesseract training data
+    """
+    mocker.patch("gi_loadouts.conf.data_prefix", f"{uuid4().hex.upper()[0:8]}-")
+    make_temp_file()
+
+    """
+    Perform the action of loading the artifact information
+    """
+    temp_prm = ""
+    tempfile = "test/static/img/gi-loadouts-ocr-test.webp"
+    savefile = str(Path(gettempdir()) / "gi-loadouts-ocr-test.webp")
+    with open(tempfile, "rb") as src_file:
+        with open(savefile, "wb") as dest_file:
+            dest_file.write(src_file.read())
+    mocker.patch.object(QFileDialog, "getOpenFileName", return_value=(savefile, temp_prm))
+    qtbot.mouseClick(scantest.arti_cnvs_load, Qt.LeftButton)
+
+    init = scantest.arti_type.currentText()
+    qtbot.waitUntil(lambda: scantest.arti_type.currentText() != init)
+
+    """
+    Perform the action of importing artifact info in MainWindow
+    """
+    qtbot.mouseClick(scantest.arti_back_done, Qt.LeftButton)
+
+    """
+    Confirm if the user interface elements change accordingly
+    """
+    assert scantest.keep_info() == __rtrn__
+
+    """
+    Clear the copied file from temp directory
+    """
+    kill_temp_file()
+    if path.exists(savefile):
+        remove(savefile)
```

Comment:
As the `NamedTemporaryFile`'s path would be reasonably randomized, we need not look for whether the path exists.

```suggestion
    remove(savefile)
```